docs/manual.rst

```diff
@@ -1337,30 +1337,31 @@ The single shapefile may also be accessed like so:
 Unsupported drivers
 -------------------
 
-In :py:attr:`fiona.supported_drivers` a selection of GDAL/OGR's
-drivers that is tested to work with Fiona is maintained. By default, Fiona 
-allows only these drivers with their listed access modes:  r for read support,
-respectively a for append and w for write.
-
-These restrictions can be circumvented by modifying :py:attr:`fiona.supported_drivers`:
+Fiona maintains a list of OGR drivers in :py:attr:`fiona.supported_drivers`
+that are tested and known to work together with Fiona. Opening a dataset using
+an unsupported driver or access mode results in an :py:attr: `DriverError`
+exception. By passing `allow_unsupported_drivers=True` to :py:attr:`fiona.open` 
+no compatibility checks are performed and unsupported OGR drivers can be used.
+However, there are no guarantees that Fiona will be able to access or write 
+data correctly using an unsupported driver.  
```

Comment:
@rbuffat this is great! You've explained the situation and the intent of the feature very well.